sdk/ai/Azure.AI.Inference/src/Customized/AIInferenceExtensions.cs

```diff
@@ -0,0 +1,58 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#nullable enable
+
+using System;
+using System.ClientModel.Primitives;
+
+namespace Azure.AI.Inference
+{
+    /// <summary>
+    /// The Azure AI Inference extensions.
+    /// </summary>
+    public static class AIInferenceExtensions
+    {
+        /// <summary>
+        /// Gets the chat completion client.
+        /// </summary>
+        /// <param name="provider"></param>
+        /// <returns></returns>
+        public static ChatCompletionsClient GetChatCompletionsClient(this ConnectionProvider provider)
+        {
+            ChatCompletionsClient chatClient = provider.Subclients.GetClient(() => CreateChatCompletionsClient(provider));
+            return chatClient;
+        }
+
+        private static ChatCompletionsClient CreateChatCompletionsClient(this ConnectionProvider provider)
+        {
+            ClientConnection connection = provider.GetConnection(typeof(ChatCompletionsClient).FullName!);
+            if (!connection.TryGetLocatorAsUri(out Uri? uri) || uri is null)
+            {
+                throw new InvalidOperationException("Invalid URI.");
+            }
+            return new ChatCompletionsClient(uri, new AzureKeyCredential(connection.ApiKeyCredential!));
```

Comment:
What if the connection is a TokenCredential connection? I think we should support it too.

Reply:
Added support for TokenCredential